specs/WebResourceRequested-CustomScheme.md

```diff
@@ -0,0 +1,322 @@
+      [in] UINT32 count,
+      [in] const ICoreWebView2CustomSchemeRegistration** schemeRegistrations);
+}
+```
+
+## .NET API
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    // Represents the registration of a custom scheme with the CoreWebView2Environment.
+    // This allows the WebView2 app to be able to handle
+    // WebResourceRequested event for requests with the specified scheme and
+    // be able to navigate the WebView2 to the custom scheme. Once the environment
+    // is created, the registrations are valid and immutable throughout the
+    // lifetime of the associated WebView2s' browser process and any WebView2
+    // environments sharing the browser process must be created with identical
+    // custom scheme registrations, otherwise the environment creation will fail.
+    // Any further attempts to register the same scheme will fail.
+    // The URIs of registered custom schemes will be treated similar to http URIs for their origins.
+    // They will have tuple origins for URIs with host and opaque origins for
+    // URIs without host as specified in [7.5 Origin - HTML Living Standard](https://html.spec.whatwg.org/multipage/origin.html)
+    // Example:
+    // `custom-scheme-with-host://hostname/path/to/resource` has origin of `custom-scheme-with-host://hostname`
+    // `custom-scheme-without-host:path/to/resource` has origin of `custom-scheme-without-host:path/to/resource`
+    // For WebResourceRequested event, the cases of request URIs and filter URIs with custom schemes
+    // will be normalized according to generic URI syntax rules. Any non-ASCII characters will be preserved.
+    runtimeclass CoreWebView2CustomSchemeRegistration
+    {
+        // Constructor
+        CoreWebView2CustomSchemeRegistration(string schemeName, bool isSecure, bool isCorsEnabled, IList<String> allowedOrigins);
+
+        // The name of the custom scheme to register.
+        string SchemeName { get; set; };
+
+        // Whether the scheme will be treated as a secure context like https.
+        bool IsSecure { get; set; };
+
+        // If `isCorsEnabled` is set, the scheme will be allowed for requests subject to
+        // CORS rules as specified in [Cross-Origin Resource Sharing - MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS).
+        // If it is not set, CORS requests with this scheme will be blocked.
+        bool IsCorsEnabled { get; set; };
+
+        // List of origins that are allowed to issue requests with the custom scheme.
+        // Except origins with custom scheme itself and no-origin requests,
+        // the origin of any request to the custom scheme URL
+        // needs to be in this list. Note that cross-origin restrictions still apply. 
+        // Origins are specified as a string in the format of scheme://host:port.
+        // The origins are string pattern matched with `*` (matches 0 or more characters) and `?` (matches 0 or 1 character) wildcards.
```

Comment:
This reminds me of the text from the syntax of the addwebresourcerequest filter. Is it the same as that? If so it would be good to say that and link to that instead of trying to define again here. That doc got updates and has a big table of examples.